sql/002b_cql.sql

```diff
@@ -321,9 +324,31 @@ BEGIN
             args := jsonb_build_array(args);
         END IF;
 
-        SELECT (queryable(a->>'property')).wrapper INTO wrapper
-        FROM jsonb_array_elements(args) a
-        WHERE a ? 'property' LIMIT 1;
+        IF jsonb_path_exists(args, '$[*] ? (@.property == "id" || @.property == "datetime" || @.property == "end_datetime" || @.property == "collection")') THEN
+            wrapper := NULL;
+        ELSE
+            -- if any of the arguments are a property, try to get the property_wrapper
+            FOR arg IN SELECT jsonb_path_query(args, '$[*] ? (@.property != null)') LOOP
+                RAISE NOTICE 'Arg: %', arg;
+                SELECT property_wrapper INTO wrapper
+                FROM queryables
+                WHERE name=(arg->>'property')
+                LIMIT 1;
+                RAISE NOTICE 'Property: %, Wrapper: %', arg, wrapper;
+                IF wrapper IS NOT NULL THEN
+                    EXIT;
+                END IF;
+            END LOOP;
+
+            -- if the property was not in queryables, see if any args were numbers
```

Comment:
I think this logic needs to be applied to a `between` op logic above. e.g.

```json
{
  "filter-lang": "cql2-json",
  "filter": {
    "op": "and",
    "args": [
      {
        "op": "=",
        "args": [
          {
            "property": "collection"
          },
          "sentinel-2-l2a"
        ]
      },
      {
        "op": "s_intersects",
        "args": [
          {
            "property": "geometry"
          },
          {
            "type": "Polygon",
            "coordinates": [
              [
                [
                  28.52298954693501,
                  28.43552029598466
                ],
                [
                  31.902934901442705,
                  28.43552029598466
                ],
                [
                  31.902934901442705,
                  32.34830434734562
                ],
                [
                  28.52298954693501,
                  32.34830434734562
                ],
                [
                  28.52298954693501,
                  28.43552029598466
                ]
              ]
            ]
          }
        ]
      },
      {
        "op": "between",
        "args": [
          {
            "property": "s2:mean_solar_azimuth"
          },
          [13000000,
          140]
        ]
      }
    ]
  },
  "fields": {
    "include": [
      "properties.s2:mean_solar_azimuth"
    ],
    "exclude": [
      "links"
    ]
  }
}
```

Returns values between 130 and 140